docs/proposal-007-apply.md

```diff
@@ -0,0 +1,137 @@
+# `eksctl apply`
+
+This proposal addresses a GitOps-compatible subcommand of `eksctl` called
+_tentatively_ `apply`. Its purpose is to subsume the various imperative commands
+`eksctl` provides now into one `apply` command that reconciles the actual state
+of an EKS cluster with the intended state as specified in the `eksctl` config file.
+
+This proposal is only concerned with the `apply` subcommand. Existing behavior
+and commands are unaffected. Continuing to support them shouldn't complicate
+the implementation too much as they are more or less subsets of
+operations performed by `apply`.
+
+## Questions to be answered
+
+-   Does the config represent the entire state of the config vs a partial
+    spec?
+    -   e.g. if I `apply` with a list of 1 nodegroups, are any other existing
+        nodegroups deleted?
```

Comment:
yes 😃 

Ideally by the end the config in general should be smarter, ie a user should be able to change any information and see it happen

_Originally posted by @Callisto13 in https://github.com/weaveworks/eksctl-private/pull/233#discussion_r552483539_

Reply:
I think so too like I wrote in the next section but it's not totally clear whether that's the right "default" (or should be the only option). For example [the `google` `terraform` provider supports IAM with 3 different levels of authoritativeness.](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_bucket_iam#google_storage_bucket_iam_policy#iam-policy-for-cloud-storage-bucket)